lib/chef/provider/service/debian.rb

```diff
@@ -70,22 +70,40 @@ def define_resource_requirements
           end
         end
 
+        def parse_init_file(path)
+          return [] unless ::File.exist?(path)
+          start = false
+          finish = false
+          ::File.readlines(path).each_with_object([]) do |line, acc|
+            if line =~ /^### BEGIN INIT INFO/
+              start = true
+              next
+            elsif line =~ /^### END INIT INFO/
+              finish = true
```

Comment:
you can probably just set start to false here and eliminate the finish boolean.  the 'next's are likely redundant as well